Sensor Library - MPU9250

JLefebvre55 opened the pull request:
Test Script Author: Angus Wu @AngusWu25 
Current Hardware Owner: Angus Wu @AngusWu25 
Final Review: Urmil Modi @urmilmodi 

1. Author reviews library contents
2. Jayden Lefebvre writes Arduino test script using library
2. Current owner tests library implementation on hardware, documents results

Comment on this PR if you have any questions

angusjhwu committed:
my sensor testing code for mpu9250

angusjhwu committed:
my sensor testing code for mpu9250

angusjhwu committed:
Added library files for mpu9250

angusjhwu committed:
Added library files for mpu9250

angusjhwu committed:
Create README.md

angusjhwu committed:
Create README.md

urmilmodi committed:
corrected repo structure and added sensor readme

urmilmodi committed:
corrected repo structure and added sensor readme

angusjhwu committed:
Put code into functions, Folders organized

angusjhwu committed:
Put code into functions, Folders organized

angusjhwu committed:
reorganized arduino code

angusjhwu committed:
forgot to commit all changes last t

angusjhwu committed:
forgot to commit all changes last t

JLefebvre55 committed:
Added Gitignore

angusjhwu committed:
Added toggle for enabling FFT and switching the target axis for FFT

angusjhwu committed:
Added toggle for enabling FFT and switching the target axis for FFT

angusjhwu committed:
Added toggle for enabling FFT and switching the target axis for FFT

angusjhwu committed:
Added toggle for enabling FFT and switching the target axis for FFT

angusjhwu committed:
Added toggle for enabling FFT and switching the target axis for FFT

angusjhwu committed:
Added toggle for enabling FFT and switching the target axis for FFT

angusjhwu committed:
Added sensor details and code description

Added the sensor measurements with units, the use of the sensor readings, and description of code. Todo: add sensor measurement ranges

angusjhwu committed:
Added sensor details and code description

Added the sensor measurements with units, the use of the sensor readings, and description of code. Todo: add sensor measurement ranges

angusjhwu committed:
cleaned up arduino code

angusjhwu committed:
cleaned up arduino code

angusjhwu committed:
Added sensor output ranges

angusjhwu committed:
Added sensor output ranges

JLefebvre55 committed:
Merge branch 'sensor-mpu9250' of https://github.com/UTHT/electronics-scripts into sensor-mpu9250

JLefebvre55 committed:
Refactored Library

JLefebvre55 committed:
Library

urmilmodi committed:
added todos

JLefebvre55 committed:
Minor Refactor

JLefebvre55 committed:
Sensor Implementation Template

JLefebvre55 committed:
Implementation

JLefebvre55 committed:
Adafruit Compatibility Fix

JLefebvre55 committed:
Restructured for Compilation

urmilmodi committed:
fixed variable declaration errors

JLefebvre55 committed:
Fixed Error Checking

Walker431 committed:
Working code for MPU9250

angusjhwu committed:
Corrected syntax

note: the last commit was also me but for some reason it used my old github account name bruh

AngusWu25 commented:
The code is working, image attached.
![working mpu9250 code](https://user-images.githubusercontent.com/78286686/120082713-796ee100-c092-11eb-89f8-d82d47aa0766.jpg)
To do: vibration analysis
Potential bugs: Serial outputs sensor values rapidly (ignoring the specified delta) after precisely 43 seconds

angusjhwu committed:
Added FFT testing printout

FFT toggle now properly implemented, and added instructions in code to guide testers on which parameters to change to enable FFT

AngusWu25 commented:
Added the necessary code for printing out the FFT outputs. Also added instructions in the .ino to guide future testers what parameters to change to enable FFT (off by default since FFT requires delta of 1, whereas it is normally 250).
Attached image shows FFT printout (removed regular sensor output printout to declutter in demo below)
![working mpu9250 FFT portion](https://user-images.githubusercontent.com/78286686/120743783-a6bef300-c4c7-11eb-9cc0-f15f49a518e9.jpg)

JLefebvre55 committed:
Added data slot for FFT

Added 10th datum for single-axis FFT analysis
Changed axis macro to double-quoted
